src/js/background.js

```diff
@@ -682,20 +714,32 @@ Badger.prototype = {
   },
 
   /**
-   * Add only new third party origins to the tabData[tabId] object for
+   * Save third party origins to tabData[tab_id] object for
    * use in the popup and, if needed, call updateBadge.
    *
-   * @param tabId the tab we are on
+   * @param tab_id the tab we are on
    * @param fqdn the third party origin to add
    * @param action the action we are taking
    *
    **/
-  logThirdPartyOriginOnTab: function(tabId, fqdn, action) {
-    if(!this.tabData[tabId].origins.hasOwnProperty(fqdn)) {
-      this.tabData[tabId].origins[fqdn] = action;
-      if (constants.BLOCKED_ACTIONS.has(action)) {
-        this.tabData[tabId].blockedCount += 1;
-        badger.updateBadge(tabId);
+  logThirdPartyOriginOnTab: function (tab_id, fqdn, action) {
```

Comment:
Could you add some tests for this? It went from 2 code paths to 4.

Reply:
Added: 31f3c741c938139022f76bcec643d9f7ce5891ba.